lib/Sema/CSApply.cpp

```diff
@@ -2029,9 +2033,6 @@ namespace {
     }
 
     Expr *handleStringLiteralExpr(LiteralExpr *expr) {
-      if (cs.getType(expr) && !cs.getType(expr)->hasTypeVariable())
-        return expr;
-      
```

Comment:
I'm a little suspect of this change, but when `@dynamicCallable` needs to make a String literal, it makes things a little awkward because we have the resulting type, but not the initializer. I'm sure there might be another way to grab it, but was unable to think of anything clever.

Reply:
I actually don't understand why these two lines were there in the first place. If the generated SIL does not change, I'm fine with this.